src/Workspaces/Core/Portable/RegularExpressions/RegexParser.cs

```diff
@@ -414,25 +419,36 @@ private RegexPrimaryExpressionNode ParseUnexpectedCloseParenToken()
         {
             var token = _currentToken.With(kind: RegexKind.TextToken).AddDiagnosticIfNone(
                 new RegexDiagnostic(WorkspacesResources.Too_many_close_parens, GetSpan(_currentToken)));
+
+            // Technically, since an error occurred, we can do whatever we want here.  However,
+            // the spirit of the native parser is that top level sequence elements are allowed
+            // to have trivia.  So that's the behavior we mimic.
             ConsumeCurrentToken(allowTrivia: true);
             return new RegexTextNode(token);
         }
 
         private RegexPrimaryExpressionNode ParseText()
+            // Allow trivia between this piece of text and the next sequence element
```

Comment:
:bulb: Commenting is good, but this layout is poor for readability. Either use a proper documentation comment on the method, or change this to a block body. (similar for other cases)

Reply:
Done.